src/js/logic/taskManager.js

```diff
@@ -6,14 +6,17 @@ export default class TaskManager {
   }
 
   #UpdateLocalStorage() {
+    this.tasksArray.forEach((tmtask, tmindex) => {
+      tmtask.index = tmindex + 1;
+    });
     localStorage.setItem('taskDbKey', JSON.stringify(this.tasksArray));
   }
 
   addTask(task) {
     if (this.tasksArray.lenght === 0) {
-      task.updateIndex(0);
+      task.updateIndex(1);
     } else {
-      task.updateIndex(this.tasksArray.lenght - 1);
+      task.updateIndex(this.tasksArray.lenght);
     }
     task.updateStatus(false);
```

Comment:
- [ ] Is this statement necessary since in your Task class you have defined a default value for the `completed` attribute to `false` for any newly created task? Let's try to keep your code DRY

Reply:
> * [ ]  Is this statement necessary? Since in your Task class you have defined a default value for the `completed` attribute to `false` for any newly created task. Let's try to keep your code DRY

Yes, it was necessary in case I was to implement the drag and drop feature.